src/bin/cobalt/new.rs

```diff
@@ -409,8 +419,26 @@ pub fn publish_document(file: &path::Path) -> Result<()> {
     let doc =
         cobalt_model::DocumentBuilder::<cobalt_model::FrontmatterBuilder>::new(front, content);
     let doc = doc.to_string();
-
     cobalt_model::files::write_document_file(doc, file)?;
 
+    if config.posts.add_date_prefix {
+        // avoid prepend to existing date prefix
+        let (_, file_name) = cobalt_model::parse_file_stem(
+            file.file_stem()
+                .map(|os| os.to_string_lossy().into_owned())
+                .unwrap_or_else(|| "".to_owned()),
+        );
+
+        let file_name = format!(
+            "{}{}.{}",
+            date.prefix(),
+            file_name,
+            file.extension()
+                .map(|os| os.to_string_lossy().into_owned())
```

Comment:
We shouldn't make an owned value to just throw it away when we can just borrow instead

Reply:
I'm really bad with lifetimes :(

```rust
file.extension()
                .map(|os| os.to_string_lossy().borrow())
```
gives me
```rust
error[E0277]: the trait bound `std::borrow::Cow<'_, str>: std::borrow::Borrow<std::string::String>` is not satisfied
   --> src\bin\cobalt\new.rs:438:48
    |
438 |                 .map(|os| os.to_string_lossy().borrow())
    |                                                ^^^^^^ the trait `std::borrow::Borrow<std::string::String>` is not implemented for `std::borrow::Cow<'_, str>`
    |
    = help: the following implementations were found:
              <std::borrow::Cow<'a, B> as std::borrow::Borrow<B>>
```

How do I specify a lifetime here?

Reply:
So `to_string_lossy` is returning a value.  You are then trying to take a reference to it but the value is a temp.

You'll need to do
```rust
let ext = file.extension().map(|os| os.to_string_lossy());
ext.map(|s| s.as_ref()).unwrap_or_else("")
```

Reply:
Doesn't work neither: 
```rust
let ex = ext.map(|os| os.as_ref());
    |                               ^^        - `os` dropped here while still borrowed
    |                               |
    |                               borrowed value does not live long enough
```